### DISCUSSION TITLE
how to share conversation_id between apps

### DISCUSSION THREAD

**Discussion by wuhongyi1977** [langgenius/dify#8054] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Help，Can conversation history(conversation_id) be shared between different chat apps?
Furthermore, can different types of conversation-enabled applications be shared(chat/agent)?

**Comment by laipz8200** [comment]
Currently, there are no plans for that. Could you let me know in what scenarios you would need such a feature?

**Reply by hllshiro** [reply]
I have a similar need: in my case, the custom tool is provided by the client itself. However, the client can access to dify server, but not the other way around. So I added a layer of proxy, which is used to bridge the dify server and the client.

![image](https://github.com/user-attachments/assets/2f293ca1-fbb2-4931-a680-1eef894bd29b)

Similar to the structure shown in the figure, the proxy registers some custom tools in dify, and when the client establishes a websocket connection to the proxy, the proxy initiates a session with dify, and binds it to the link after getting the conversation_id. Later, when dify initiates functionCalling, the proxy forwards it to the client for execution.

**Reply by wuhongyi1977** [reply]
A normal chat application and an application (agent) with search and drawing are created, and then it is necessary to mix the normal chat application and the agent application in a user session. I feel that using the agent application alone will cause additional overhead. After all, there are more scenarios where there is no need to call tools.
